### Idea: New bus route has similar travel or even more time compared to the old ones. 

### Null Hypothesis $H_0$: 
New bus route average travel time will be greater than or equal to that of the old routes.
### $H_0$: T(new) >= T(old)


### Hypothesis $H_a$:
New bus route average travel time will be less than that of the old routes.
### $H_a$: T(new) < T(old)

### $a = 0.05$

In [2]:
# PUI2018 - Homework 4 - Gabriel Yin - ty1045 - N10088627 
author = "__Gabriel__"
contact = "__gabrielyin@berkeley.edu__"
# import packages
import pandas as pd 
import numpy as np 
import requests, json
import matplotlib.pyplot as plt 
import seaborn as sns 

In [29]:
# download raw data 
!wget https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt

--2018-10-03 00:36:21--  https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.200.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.200.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1179 (1.2K) [text/plain]
Saving to: ‘times.txt’

times.txt           100%[===================>]   1.15K  --.-KB/s    in 0s      

2018-10-03 00:36:21 (164 MB/s) - ‘times.txt’ saved [1179/1179]



In [34]:
# read into dataframe 
raw_data = pd.read_csv("times.txt", header=None)

In [35]:
# print dataframe statistics
raw_data.info()
raw_data.describe()
raw_data.head(10)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 1 columns):
0    100 non-null float64
dtypes: float64(1)
memory usage: 880.0 bytes


0
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550
5  34.822079
6  39.871884
7  39.579946
8  31.026587
9  27.662461

In [40]:
# z_score calculation
MU, TIME, LEN, SIGMA = 36.0, float(raw_data.mean()), len(raw_data), float(raw_data.std())
calc_z = lambda time, mu, sigma, n : (time - mu) / (sigma / (np.sqrt(n)))
z_score = calc_z(TIME, MU, SIGMA, LEN)
print("Calculated z-score: %f Abs z-score: %f" % (z_score, abs(z_score)))

Calculated z-score: -2.149052 Abs z-score: 2.149052


### In this case, since z-score, 2.149, is greater than the critical value at 0.05, which is 1.96. Therefore, we could thus reject the null hypothesis. We can say that the new bus route average travel time is faster than the old bus route travel time.